src/App.tsx

```diff
@@ -1,55 +1,104 @@
-import React from 'react';
+import {
+  ChangeEvent,
+  FC,
+  MouseEvent,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import cn from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropDownContent } from './components/DropDownContent/DropDownContent';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = <T extends (...args: never[]) => void>(
+  func: T,
+  delay: number,
+) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: Parameters<T>): void => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
+
+export const App: FC = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [query, setQuery] = useState('');
+  const [isSelectedQuery, setIsSelectedQuery] = useState(false);
```

Comment:
Let's create one more `useState` with `appliedQuery` and rename boolean `useState`.
Boolean `useState` we will use to render our list.
`appliedQuery` we will use for creating our `visiblePeople`

```suggestion
  const [query, setQuery] = useState('');
  const [appliedQuery, setAppliedQuery] = useState('');
  const [isSuggestionsActive, setIsSuggestionsActive] = useState(false);
```